clang/lib/Sema/SemaStmt.cpp

```diff
@@ -3747,6 +3747,16 @@ Sema::ActOnReturnStmt(SourceLocation ReturnLoc, Expr *RetValExp,
         Diag(ReturnLoc, diag::err_acc_branch_in_out_compute_construct)
         << /*return*/ 1 << /*out of */ 0);
 
+  // using plain return in a coroutine is not allowed.
+  FunctionScopeInfo *FSI = getCurFunction();
+  if (getLangOpts().Coroutines && FSI->isCoroutine()) {
```

Comment:
Could we only do this when `FSI->FirstReturnLoc.isInvalid() == true` to make sure we do not start producing the error on every return statement?

Previously we only produced a single error, and this seems like a useful behavior to keep. Otherwise we're spamming people with error messages.

Reply:
Done.